cmd/collector/app/server/grpc.go

```diff
@@ -42,31 +40,20 @@ type GRPCServerParams struct {
 // StartGRPCServer based on the given parameters
 func StartGRPCServer(params *GRPCServerParams) (*grpc.Server, error) {
 	var server *grpc.Server
-	var grpcOpts []grpc.ServerOption
-
-	if params.MaxReceiveMessageLength > 0 {
-		grpcOpts = append(grpcOpts, grpc.MaxRecvMsgSize(params.MaxReceiveMessageLength))
-	}
-	grpcOpts = append(grpcOpts, grpc.KeepaliveParams(keepalive.ServerParameters{
-		MaxConnectionAge:      params.MaxConnectionAge,
-		MaxConnectionAgeGrace: params.MaxConnectionAgeGrace,
-	}))
-
-	if params.TLSConfig.Enabled {
-		// user requested a server with TLS, setup creds
-		tlsCfg, err := params.TLSConfig.Config(params.Logger)
-		if err != nil {
-			return nil, err
-		}
-
-		creds := credentials.NewTLS(tlsCfg)
-		grpcOpts = append(grpcOpts, grpc.Creds(creds))
+	var grpcOpts []configgrpc.ToServerOption
+
+	server, err := params.ToServer(context.Background(), nil, component.TelemetrySettings{
+		Logger: params.Logger,
+		LeveledMeterProvider: func(_ configtelemetry.Level) metric.MeterProvider {
+			return noop.NewMeterProvider()
+		},
```

Comment:
We can construct a `telemetry.NoopSettings` and then set the logger. It might even be nice to add a helper similar to `FromOtelComponent` that does a `ToOtelComponent`